src/bindings/jna/libelektra/src/main/java/org/libelektra/ReadableKey.java

```diff
@@ -0,0 +1,480 @@
+	 * @return
+	 *         <ul>
+	 *         <li>0 if key name is equal</li>
+	 *         <li>-1 if this key name has lower alphabetical order than the
+	 *         {@code other} key</li>
+	 *         <li>1 if this key has higher alphabetical order</li>
+	 *         </ul>
+	 * @throws IllegalStateException    if this or the {@code other}
+	 *                                  {@link ReadableKey} has already been
+	 *                                  released
+	 * @throws IllegalArgumentException if {@code other} is {@code null}
+	 */
+	@Override public int compareTo (ReadableKey other)
+	{
+		argNotNull (other, "Key 'other'");
+		return Integer.signum (Elektra.INSTANCE.keyCmp (getPointer (), other.getPointer ()));
+	}
+
+	@Override public int hashCode ()
+	{
+		return Objects.hashCode (getName ());
+	}
+
+	@Override public boolean equals (Object obj)
+	{
+		if (this == obj)
+		{
+			return true;
+		}
+		if (obj == null || getClass () != obj.getClass ())
+		{
+			return false;
+		}
+		ReadableKey other = (ReadableKey) obj;
+		return this.compareTo (other) == 0;
+	}
+
+	/**
+	 * Checks whether this key is sub-key of the {@code other} key
+	 *
+	 * @param other Key that is used in check as parent key
+	 * @return Boolean if this key is (non-direct) sub-key of other-key
+	 * @throws IllegalStateException    if this or the {@code other}
+	 *                                  {@link ReadableKey} has already been
+	 *                                  released
+	 * @throws IllegalArgumentException if {@code other} is {@code null}
+	 */
+	public boolean isBelow (ReadableKey other)
+	{
+		argNotNull (other, "Key 'other'");
+		return Elektra.INSTANCE.keyIsBelow (other.getPointer (), getPointer ()) == 1;
+	}
+
+	/**
+	 * Checks whether this key is the same as the {@code other} key or a sub-key of
+	 * the {@code other} key
+	 *
+	 * @param other Key that is used in check as parent key
+	 * @return Boolean if this key is other key or (non-direct) sub-key of other-key
+	 * @throws IllegalStateException    if this or the {@code other}
+	 *                                  {@link ReadableKey} has already been
+	 *                                  released
+	 * @throws IllegalArgumentException if {@code other} is {@code null}
+	 */
+	public boolean isBelowOrSame (ReadableKey other)
+	{
+		argNotNull (other, "Key 'other'");
+		return Elektra.INSTANCE.keyIsBelowOrSame (other.getPointer (), getPointer ()) == 1;
+	}
+
+	/**
+	 * Checks whether this key is direct sub-key of the {@code other} key
+	 *
+	 * @param other Key that is used in check as parent key
+	 * @return Boolean if this key is direct sub-key of other key ("child")
+	 * @throws IllegalStateException    if this or the {@code other}
+	 *                                  {@link ReadableKey} has already been
+	 *                                  released
+	 * @throws IllegalArgumentException if {@code other} is {@code null}
+	 */
+	public boolean isDirectBelow (ReadableKey other)
```

Comment:
Typo or is the C API naming not suitable?
```suggestion
	public boolean isDirectlyBelow (ReadableKey other)
```

Reply:
Thank you, i will include this change in PR #4002